app-modules/campaign/database/migrations/2024_11_15_054949_data_fill_created_by.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use AdvisingApp\Campaign\Models\Campaign;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        Campaign::with('user')->each(function ($campaign) {
+            $campaign->created_by_id = $campaign->user?->getKey();
+            $campaign->created_by_type = 'user';
+            $campaign->save();
+        });
```

Comment:
Please review the documentation on data migrations: https://github.com/canyongbs/advisingapp/blob/main/docs/data-migrations.md

Data migrations cannot contain references to classes like `Campaign`. To make changes in the database, you must use the DB facade.

Let's also make sure we have a fall back here just in case the campaign row does not have a `user_id` set and get the first record non-deleted record from the `users` table.

Reply:
@Orrison 
What do you mean by "get the first record non-deleted record from the users table."? Can you please explain little bit here.

Reply:
I meant to get the actual first User in the database that was not soft deleted to use in case the `user_id` column was `null`